dll/win32/winmm/time.c

```diff
@@ -438,7 +447,11 @@ MMRESULT WINAPI timeBeginPeriod(UINT wPeriod)
 MMRESULT WINAPI timeEndPeriod(UINT wPeriod)
 {
     if (wPeriod < MMSYSTIME_MININTERVAL || wPeriod > MMSYSTIME_MAXINTERVAL)
-	return TIMERR_NOCANDO;
+		return TIMERR_NOCANDO;
+
+	/*  High resolution timer request finished, stop using QPC */
+	if (wPeriod <= 5 && TIME_qpcFreq.QuadPart != 0)
+		TIME_qpcFreq.QuadPart = 0;
```

Comment:
Do we need to check `wPeriod` here?

Reply:
Actually, `QueryPerformanceFrequency` result does not change across reboots so I don't see a reason to clear it

Reply:
Technically there should be a stack of periods that we remember, and we only set it back to zero if the stack is empty. But this seems fine to start with.

Reply:
QueryPerformanceFrequency is used both to cache the frequency and as a flag to indicate weather QueryPerformanceFrequency should be used or GetTickCount, if you set it to 0 it reverts back to GetTickCount.

Reply:
Additionally the official timeGetTime affects the system scheduler and the system as a whole, whereas this patch is application local.